Automattic/jetpack
# Link Manager: Port "Check once rather than always force on" to jetpack_mu_wpcom

Related: https://github.com/Automattic/dotcom-forge/issues/6280

## Proposed changes:

* This PR ports the existing code from WPCOMSH in https://github.com/Automattic/wpcomsh/pull/1690 to `jetpack_mu_wpcom` so it can be re-used on both Simple and Atomic sites.
* There is a detailed summary of why this port and update is happening in this associated diff: D144570-code

### Other information:


## Jetpack product discussion


Slack discussion: p1712347061956749-slack-C06DN6QQVAQ

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:






* Test for Simple sites using these instructions: D144570-code
* Test for Atomic sites using these instructions: https://github.com/Automattic/wpcomsh/pull/1690
* Ensure that this change does not cause regressions on "default" interface sites that still use nav-unification.

Note that to roll this out there will be a bit of time on Atomic sites where `wpcomsh_maybe_enable_link_manager` in WPCOMSH and `wpcom_maybe_enable_link_manager` (introduced here) will exist and run at the same time. I do not suspect that this will be an issue, but please let me know if you think it is a concern.
